Ntuplizer/plugins/JetsNtuplizer.cc

```diff
@@ -79,37 +79,81 @@ bool JetsNtuplizer::looseJetID( const pat::Jet& j ) {
+//  int npr    = chMult + neMult;
+//  int NumConst = npr;
+//
+//  return (nhf<0.99 && nemf<0.99 && NumConst>1 && muf < 0.8) && ((fabs(eta) <= 2.4 && chf>0 && chMult>0 && cemf<0.99) || fabs(eta)>2.4);      
+
+  // Change to 13 TeV definition
+  // https://twiki.cern.ch/twiki/bin/view/CMS/JetID#Recommendations_for_13_TeV_data
+
   double eta = j.eta();		
   double chf = j.chargedHadronEnergyFraction();
   double nhf = j.neutralHadronEnergyFraction(); // + j.HFHadronEnergyFraction();
-  double muf = j.muonEnergy()/(j.jecFactor(0) * j.energy());  
   double nemf = j.neutralEmEnergyFraction();
   double cemf = j.chargedEmEnergyFraction();
   int chMult = j.chargedMultiplicity();
   int neMult = j.neutralMultiplicity();
   int npr    = chMult + neMult;
   int NumConst = npr;
 
-  return (nhf<0.99 && nemf<0.99 && NumConst>1 && muf < 0.8) && ((fabs(eta) <= 2.4 && chf>0 && chMult>0 && cemf<0.99) || fabs(eta)>2.4);        
+  if(abs(eta) <= 2.7){
+    return (nhf<0.99 && nemf<0.99 && NumConst>1) && ((abs(eta)<=2.4 && chf>0 && chMult>0 && cemf<0.99) || abs(eta)>2.4);
+  }else if(abs(eta) <= 3.0){
+    return (nemf<0.90 && neMult>2);
+  }else{
+    return (nemf<0.90 && neMult>10);
+  }
+
       		
 }
 
 //===================================================================================================================
 bool JetsNtuplizer::tightJetID( const pat::Jet& j ) {
 
   //In sync with: https://twiki.cern.ch/twiki/bin/viewauth/CMS/JetID#Recommendations_for_8_TeV_data_a
+//  double eta = j.eta();		
+//  double chf = j.chargedHadronEnergyFraction();
+//  double nhf = j.neutralHadronEnergyFraction(); // + j.HFHadronEnergyFraction();
+//  double muf = j.muonEnergy()/(j.jecFactor(0) * j.energy());  
+//  double nemf = j.neutralEmEnergyFraction();
+//  double cemf = j.chargedEmEnergyFraction();
+//  int chMult = j.chargedMultiplicity();
+//  int neMult = j.neutralMultiplicity();
+//  int npr    = chMult + neMult;
+//  int NumConst = npr;
+//     
+//  return (nhf<0.90 && nemf<0.90 && NumConst>1 && muf<0.8) && ((fabs(eta)<=2.4 && chf>0 && chMult>0 && cemf<0.90) || fabs(eta)>2.4);  		
+  // Change to 13 TeV definition
+  // https://twiki.cern.ch/twiki/bin/view/CMS/JetID#Recommendations_for_13_TeV_data
+
   double eta = j.eta();		
   double chf = j.chargedHadronEnergyFraction();
   double nhf = j.neutralHadronEnergyFraction(); // + j.HFHadronEnergyFraction();
-  double muf = j.muonEnergy()/(j.jecFactor(0) * j.energy());  
   double nemf = j.neutralEmEnergyFraction();
   double cemf = j.chargedEmEnergyFraction();
   int chMult = j.chargedMultiplicity();
   int neMult = j.neutralMultiplicity();
   int npr    = chMult + neMult;
   int NumConst = npr;
-     
-  return (nhf<0.90 && nemf<0.90 && NumConst>1 && muf<0.8) && ((fabs(eta)<=2.4 && chf>0 && chMult>0 && cemf<0.90) || fabs(eta)>2.4);  		
+
+  if(abs(eta) <= 2.7){
+    return (nhf<0.90 && nemf<0.90 && NumConst>1) && ((abs(eta)<=2.4 && chf>0 && chMult>0 && cemf<0.99) || abs(eta)>2.4);
```

Comment:
For the diboson analyses, we prefer to have the "TightLepVeto" ID with tighter muon and electron cuts rather than "Tight", since we are very sensitive to such kind of noise.